bin/analyzeShearTest.py

```diff
@@ -0,0 +1,265 @@
+#
+# This product includes software developed by the
+# LSST Project (http://www.lsst.org/).
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.    See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <http://www.lsstcorp.org/LegalNotices/>.
+#
+import argparse
+import math
+import os.path
+
+import lsst.daf.persistence as dafPersist
+import lsst.afw.table as afwTable
+"""
+analyzeShearTest is a portion of the shear test suite which summarizes the information
+from all of the GalSim runs and creates a table which is suitable for plotting.
+
+The data is divided into GalSim runs each with a filter[23], seeing[.05 .07 .09] and shear.
+All subfields in the run have the same shear, but each has a randomly chosen shear angle.
+In this program, we average each subfield to get an estimator of the shear value.  In our
+initial tests, there are 32 subfields with a different shear angle.  There are 1024 galaxies
+for each subfield, comprised of 512 galaxy pair rotated at 90 degrees to each other.
+
+These 32 samples are averaged to provide an overall estimator of the shear, and the stddev of
+these samples provides an error for this estimator.
+"""
+def getCatalog(catName):
+    #  This routine processes one run at a time, and appends to an existing catalog.
+    if os.path.exists(catName):
+        outCat = afwTable.BaseCatalog.readFits(catName)
+    else:
+    #   if the catalog has not been created yet, create an empy one with the right schema.
+        schema = afwTable.Schema()
+        schema.addField("filter", type = int, doc = "")
+        schema.addField("seeing", type = float, doc = "")
+        schema.addField("shear", type = float, doc = "")
+        schema.addField("nsub", type = int, doc = "")
+        schema.addField("nsource", type = int, doc = "")
+        schema.addField("g", type = float, doc = "")
+        schema.addField("eAvg", type = float, doc = "")
+        schema.addField("eStd", type = float, doc = "")
+        schema.addField("e1Dev", type = float, doc = "")
+        schema.addField("e1Std", type = float, doc = "")
+        schema.addField("e2Dev", type = float, doc = "")
+        schema.addField("e2Std", type = float, doc = "")
+        outCat = afwTable.BaseCatalog(schema)
+    return outCat
+
+if __name__ == "__main__":
+    """
+
+    base_dir         Name of the subdirectory containing run_params file
+    out_cat          Name of the catalog for output. .fits is appended
+    """
```

Comment:
A docstring here doesn't actually do any good, because it's just an `if` statement, not a function.  I think it's redundant with the help kwargs to `parser.add_argument` below.